packages/gatsby-cli/src/structured-errors/error-map.ts

```diff
@@ -474,15 +474,21 @@ const errors = {
   },
 }
 
-export type ErrorId = keyof typeof errors
+export type ErrorId = string | keyof typeof errors
```

Comment:
The point of this type is to make sure all used error codes are part of this `errors` object. By adding `string` to the type, I think it essentially moots the whole point of the type, as `string` would subsume the `keyof typeof errors` part.

What was the original reason for needing to add `string` here and is there another way of achieving the same?

I suspect the "proper" approach would be to create a subclass of the `ErrorType`, like an `ExtendedErrorType`, which would be a union between the original type and an arbitrarily extended type etc. Probably very involved.